# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1690705303 -7200
# Parent  dbf3419ad21f055c8396d072156b84301c4ab66f
Feature: Reduce row height in multi-line view to match height of cards.

diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -4795,25 +4795,25 @@ var threadPane = {
    */
   densityChange() {
     // The class ThreadRow can't be referenced because it's declared in a
     // different scope. But we can get it from customElements.
     let rowClass = customElements.get("thread-row");
     let cardClass = customElements.get("thread-card");
     switch (UIDensity.prefValue) {
       case UIDensity.MODE_COMPACT:
-        rowClass.ROW_HEIGHT = rowClass.MULTILINE ? 36 : 18;
+        rowClass.ROW_HEIGHT = rowClass.MULTILINE ? 34 : 18;
         cardClass.ROW_HEIGHT = 40;
         break;
       case UIDensity.MODE_TOUCH:
-        rowClass.ROW_HEIGHT = rowClass.MULTILINE ? 64 : 32;
+        rowClass.ROW_HEIGHT = rowClass.MULTILINE ? 52 : 32;
         cardClass.ROW_HEIGHT = 52;
         break;
       default:
-        rowClass.ROW_HEIGHT = rowClass.MULTILINE ? 52 : 26;
+        rowClass.ROW_HEIGHT = rowClass.MULTILINE ? 46 : 26;
         cardClass.ROW_HEIGHT = 46;
         break;
     }
   },
 
   /**
    * Store the current thread tree selection.
    */
diff --git a/mail/themes/shared/mail/threadPane.css b/mail/themes/shared/mail/threadPane.css
--- a/mail/themes/shared/mail/threadPane.css
+++ b/mail/themes/shared/mail/threadPane.css
@@ -388,26 +388,26 @@ tr[is="thread-row"][multiline="true"] td
 }
 :root[uidensity="compact"] tr[is="thread-row"][multiline="true"] td {
   padding-top: 2px;
 }
 :root[uidensity="compact"] tr[is="thread-row"][multiline="true"] td.subjectcol-column {
   top: 7px;
 }
 tr[is="thread-row"][multiline="true"] td {
-  padding-top: 8px;
+  padding-top: 6px;
 }
 tr[is="thread-row"][multiline="true"] td.subjectcol-column {
-  top: 10px;
+  top: 9px;
 }
 :root[uidensity="touch"] tr[is="thread-row"][multiline="true"] td {
-  padding-top: 14px;
+  padding-top: 8px;
 }
 :root[uidensity="touch"] tr[is="thread-row"][multiline="true"] td.subjectcol-column {
-  top: 12px;
+  top: 10px;
 }
 
 /* Make the subject column header fixed width and remove the splitter */
 table[is="tree-view-table"][multiline="true"] #subjectCol {
   width: 12px !important;
   min-width: 12px !important;
   border: 1px solid var(--splitter-color);
 }
